### PR TITLE
fix: do not create empty batches in ingester

### PR DESCRIPTION
**Pull request by crepererum**
This would result in `QueryableBatch`es w/ empty schemas which results no primary keys and empty sort keys, and finally the whole thing panics w/ `Super sort key is empty`.

### CODE REVIEW

File: ingester/src/data.rs [hunk 1]
```diff
@@ -1027,13 +1027,16 @@ impl PartitionData {
...
-            return;
-        }
+            .snapshot_to_queryable_batch(table_name, Some(tombstone.clone()))
+        {
+            Some(query_batch) if !query_batch.is_empty() => query_batch,
+            _ => {
+                // No need to procedd further
+                return;
+            }
+        };
```
Comment (NGA-TRAN) [inline ingester/src/data.rs:1039]: Does it mean we also return if ` Some(query_batch)` AND `query_batch.is_empty()`?

**Review by NGA-TRAN — APPROVED** [review]
Look good if the answer of my question is a YES

File: ingester/src/data.rs (hunk 1 above)
Comment (crepererum) [reply ingester/src/data.rs:1039]: yes

**Comment by alamb** [comment]
I restartd https://app.circleci.com/pipelines/github/influxdata/influxdb_iox/18607/workflows/6690307d-1d64-4f9d-a4be-d7508c39fee8/jobs/155605 due to seemingly intermittent failing in end to end tests

File: ingester/src/data.rs [hunk 2]
```diff
@@ -1027,13 +1027,16 @@ impl PartitionData {
...
-            .snapshot_to_queryable_batch(table_name, Some(tombstone.clone()));
-        if query_batch.is_empty() {
-            //No need to procedd further
-            return;
-        }
+            .snapshot_to_queryable_batch(table_name, Some(tombstone.clone()))
+        {
+            Some(query_batch) if !query_batch.is_empty() => query_batch,
+            _ => {
+                // No need to procedd further
```
Comment (alamb) [inline ingester/src/data.rs:1036]:
```suggestion
                // No need to proceed further
```

**Pull request merged**